test/frontdesk_test.rb

```diff
@@ -0,0 +1,161 @@
+
+    it "returns an array with length equal to the number of available rooms" do
+      18.times do |i|
+        reservation = Hotel::Reservation.new(
+          rooms: Hotel::Room.new(i+1), 
+          date_range: @range)
+        @front_desk.reserve_room(@range)
+      end
+      expect(@front_desk.get_available_rooms(@range).length).must_equal 2
+    end
+
+    it "returns an array with length 20 if no reservations made" do
+      expect(@front_desk.get_available_rooms(@range).length).must_equal 20
+    end
+
+    it "returns the first room if there is no reservation made for it" do
+      expect(@front_desk.get_available_rooms(@range).first.number).must_equal 1
+    end
+  end
+
+  describe "#reserve_room" do
+    it "returns an instance of a new Reservation" do
+      expect(@front_desk.reserve_room(@range)).must_be_instance_of Hotel::Reservation
+    end
+
+    it "returns a new Reservation" do
+      check_in = Date.parse("1 Apr 2020")
+      check_out = Date.parse("3 Apr 2020")
+      expect(@front_desk.reserve_room(@range).rooms.first).must_be_instance_of Hotel::Room
+      expect(@front_desk.reserve_room(@range).date_range.start_date).must_equal check_in
+      expect(@front_desk.reserve_room(@range).date_range.end_date).must_equal check_out
+      expect(@front_desk.reserve_room(@range).cost).must_equal 400
+    end
+
+    it "adds a new Reservation to the list of reservations" do
+      check_in = Date.parse("1 Apr 2020")
+      check_out = Date.parse("3 Apr 2020")
+      @front_desk.reserve_room(@range)
+      expect(@front_desk.reservations.size).must_equal 1
+      expect(@front_desk.reserve_room(@range).rooms.first.number).must_equal 2
+      expect(@front_desk.reserve_room(@range).rooms.first).must_be_instance_of Hotel::Room
+      expect(@front_desk.reserve_room(@range).date_range.start_date).must_equal check_in
+      expect(@front_desk.reserve_room(@range).date_range.end_date).must_equal check_out
+      expect(@front_desk.reserve_room(@range).cost).must_equal 400
+    end
+
+    it "raises and Argument Error if there are no available rooms for provided dates" do
+      20.times do
+        @front_desk.reserve_room(@range)
+      end
+      expect{@front_desk.reserve_room(@range)}.must_raise ArgumentError
+    end
+
+    it "does not let reserve a room if there is dates overlapping for 2 reservations" do
+      range2 = Hotel::DateRange.new(start_date: "1 Apr 2020", end_date: "5 Apr 2020")
+      20.times do
+        @front_desk.reserve_room(@range)
+      end
+      expect{@front_desk.reserve_room(range2)}.must_raise ArgumentError
+    end
+
+    it "reserves a room if check_in and check-out are the same dates for 2 reservations" do
+      range2 = Hotel::DateRange.new(start_date: "3 Apr 2020", end_date: "7 Apr 2020")
+      20.times do
+        @front_desk.reserve_room(@range)
+      end
+      expect(@front_desk.reserve_room(range2)).must_be_instance_of Hotel::Reservation
+    end
+  end
+
+  describe "#reserve_block" do
```

Comment:
Nice work on these methods and tests. Note, you should also test that you can't reserve a block for rooms that have already been reserved.